### PR TITLE
Can not store files in SCM bridged packages

### PR DESCRIPTION
**Pull request by hennevogel**
They are maintained on the SCM...

### CODE REVIEW

**Comment by hennevogel** [comment]
@adrianschroeter should something happen with existing files if you "convert" a package to scmsync?

**Comment by hennevogel** [comment]
> should something happen with existing files if you "convert" a package to scmsync?

I see, the service is just removing them...

**Comment by adrianschroeter** [comment]
it would be good if the error message contains the URL to the scm, that would allow the user to work on the sources directly

**Comment by hennevogel** [comment]
> it would be good if the error message contains the URL to the scm, that would allow the user to work on the sources directly

It now always present and file editing isn't offered anymore.

Screenshot from 2022-03-04 16-48-40

File: src/api/app/models/package.rb [hunk 1]
```diff
@@ -1161,6 +1161,8 @@ def patchinfo
   end
 
   def delete_file(name, opt = {})
+    raise ScmsyncReadOnly if scmsync.present?
```
Comment (adrianschroeter) [inline src/api/app/models/package.rb:1164]: I would still handover the scm url to the user here, so osc users have a chance to see where they can proceed.

File: src/api/app/models/package.rb [hunk 2]
```diff
@@ -1236,6 +1238,8 @@ def self.verify_file!(pkg, name, content)
   end
 
   def save_file(opt = {})
+    raise ScmsyncReadOnly if scmsync.present?
```
Comment (adrianschroeter) [inline src/api/app/models/package.rb:1241]: same here

File: src/api/app/models/service.rb [hunk 3]
```diff
@@ -82,6 +83,8 @@ def add_service(name, parameters = [], mode = nil)
   end
 
   def save
+    raise ScmsyncReadOnly if package.scmsync.present?
```
Comment (adrianschroeter) [inline src/api/app/models/service.rb:86]: and here

**Review by adrianschroeter — APPROVED** [review]
but also fine for me that way.

**Comment by hennevogel** [comment]
hmmm

File: src/api/app/models/package.rb (hunk 1 above)
Comment (krauselukas) [reply src/api/app/models/package.rb:1167]: @adrianschroeter Those methods are not used for the API. I added it to `source_controller.rb` and also provide the SCM URL.

**Comment by hennevogel** [comment]
coveralls hangs. codecov is fine. merging!

**Pull request merged**